test/composition/slot-to-content.html

```diff
@@ -0,0 +1,110 @@
+<!doctype html>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1.0, user-scalable=yes">
+
+    <script src="../../../webcomponentsjs/webcomponents.js"></script>
+    <script src="../../../web-component-tester/browser.js"></script>
+    <script src="../helpers/chainable-methods-for-partial.js"></script>
+
+    <!-- Step 1: import the element to test -->
+    <link rel="import" href="../../starcounter-include.html">
+</head>
+
+<body>
+    <test-fixture id="default-composition">
+        <template>
+            <starcounter-include partial="{
+                &quot;LauncherLayoutInfo&quot;: {
+                    &quot;PartialId&quot;: &quot;given PartialId&quot;
+                },
+                &quot;Html&quot;: &quot;../template_w_declarative-shadow-dom_and_slots.html&quot;,
+                &quot;doesItWork&quot;: &quot;works!&quot;
+            }"></starcounter-include>
+        </template>
+    </test-fixture>
+    <template id="reference-composition">
+		<h2>Custom Shadow DOM composition</h2>
+		<content name="my-slot" select="[slot='my-slot']"></content>
+		<content name='my-slot' select="[slot='my-slot']"></content>
+		<content></content>
+		<content name="my-slot" select="[slot='my-slot']"></content>
+		<content></content>
+		<content foo="bar" name="my-slot" baz select="[slot='my-slot']"></content>
+    </template>
+
+    <script id="toBeLoadedAsString" type="text/html">
+        <h2>Custom Shadow DOM composition</h2>
+		<slot name="my-slot"></slot>
+		<slot name='my-slot'></slot>
+		<slot></slot>
+		<slot    name="my-slot"    ></slot>
+		<slot       ></slot   >
+		<slot foo="bar" name="my-slot" baz></slot>
```

Comment:
I think we need one more case for the comment above. _left side white space in closing tag_.